src/Voron/Data/BTrees/Tree.cs

```diff
@@ -1552,6 +1602,7 @@ public ValueReader GetValueReaderFromHeader(TreeNodeHeader* node)
         public void Rename(Slice newName)
         {
             Name = newName;
+            State.Modify();
```

Comment:
This is actually a special case, because we are going to be rename the value reader and we have to force the state to be considered as Modified even though we are not going to change anything in it. I will add a comment in there.